### PR TITLE
feat(types): add nuxt error type

### PR DESCRIPTION
**Pull request by orblazer**
## Types of changes

- [ ] Bug fix (a non-breaking change which fixes an issue)
- [x] New feature (a non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Description

This add type for Nuxt error with interface `NuxtError`.
With that on error layout we can now use :
```ts
import { Component, Vue, Prop } from 'nuxt-property-decorator'
import { NuxtError } from '@nuxt/vue-app'

@Component({})
export default class ErrorPage extends Vue {
  @Prop({ required: true }) readonly error!: NuxtError;
}
```

## Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly. (PR: #)
- [ ] I have added tests to cover my changes (if not applicable, please state why)
- [x] All new and existing tests are passing.

### CODE REVIEW

**Comment by kevinmarrec** [comment]
@orblazer Isn't it related to https://github.com/nuxt/nuxt.js/blob/dev/packages/vue-app/types/index.d.ts#L71 ?

**Comment by orblazer** [comment]
Yes @kevinmarrec we probably can merge the 2 interface et keep the name `NuxtError`

**Comment by kevinmarrec** [comment]
@orblazer Alright, agreed. I would just want to know how you found the `path` property if you can link me code or documentation. Thanks in advance.

**Comment by orblazer** [comment]
@kevinmarrec I have found `path` here : https://github.com/nuxt/nuxt.js/blob/31f5729828e1009a6db28fe90296c178451f9efd/packages/vue-app/template/server.js#L83

**Comment by kevinmarrec** [comment]
@orblazer Alright, then I'd say we can remove `ErrorParams` in favor of  `NuxtError`. Then do
```ts
export interface NuxtError {
  message?: string
  path?: string
  statusCode?: number
}
```
> Comparing to your changes, notice `statusCode` as `number` and the alphabetical order of properties.

And also change `Context.error` signature to
```ts
error(error: NuxtError): void
```

File: packages/vue-app/types/index.d.ts [hunk 1]
```diff
@@ -68,9 +68,10 @@ export interface Transition {
   leaveCancelled?(el: HTMLElement): void
 }
 
-export interface ErrorParams {
-  statusCode?: number
+export interface NuxtError {
   message?: string
+  statusCode?: number
+  path?: string
```
Comment (kevinmarrec) [inline packages/vue-app/types/index.d.ts]: alphebtical order
```
message
path
statusCode
```

**Review by kevinmarrec — CHANGES_REQUESTED** [review]

File: packages/vue-app/types/index.d.ts (hunk 1 above)
Comment (orblazer) [reply packages/vue-app/types/index.d.ts]: Its good, i have miss re order property ;)

Comment (kevinmarrec) [reply packages/vue-app/types/index.d.ts]: 😁

**Review by kevinmarrec — APPROVED** [review]

**Comment by kevinmarrec** [comment]
LGTM, thanks for the types improvement @orblazer

**Pull request merged**